### PR TITLE
ROE-1705 Validate public register jurisdiction field

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link
https://companieshouse.atlassian.net/browse/ROE-1705

### Change description

AC 5 - check not empty
AC 7 - check valid characters
AC 8 - check combined length of name and jurisdiction are less than 160 characters - based on work for dual nationality.

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/validation/custom.validation.ts [hunk 1]
```diff
@@ -97,6 +97,14 @@ export const checkSecondNationality = (nationality: string = "", secondNationali
   return true;
 };
 
+export const checkPublicRegisterJurisdiction = (public_register_name: string = "", public_register_jurisdiction: string = "") => {
+  const separator = ",";
+  if (public_register_name && public_register_jurisdiction && `${public_register_name}${separator}${public_register_jurisdiction}`.length > 160) {
```
Comment (mwestacott) [inline src/validation/custom.validation.ts:102]: I can't see any new testing around this.

I'd expect tests when total length is exceeded and not exceeded. When one of the fields is over 158 characters (presume other must be at least 1 + comma) and another when both are over 160, plus any other scenarios you can think of.

**Review by mwestacott — COMMENTED** [review]
Please see comments. Trust you've tested this in Docker too.

File: src/validation/error.messages.ts [hunk 2]
```diff
@@ -97,6 +99,7 @@ export enum ErrorMessages {
   MAX_COUNTY_LENGTH = "County, state, province or region must be 50 characters or less",
   MAX_POSTCODE_LENGTH = "Postcode must be 15 characters or less",
   MAX_ENTITY_PUBLIC_REGISTER_NAME_LENGTH = "Name of register must be 160 characters or less",
```
[the remark this answers is not in the document]
Comment (mattch1) [reply src/validation/error.messages.ts]: Not sure from reading the ticket its not included in the AC so I guess we can remove it.

File: src/validation/custom.validation.ts [hunk 3]
```diff
@@ -97,6 +97,14 @@ export const checkSecondNationality = (nationality: string = "", secondNationali
   return true;
 };
 
+export const checkPublicRegisterJurisdiction = (public_register_name: string = "", public_register_jurisdiction: string = "") => {
+  const separator = ",";
```
[the remark this answers is not in the document]
Comment (mattch1) [reply src/validation/custom.validation.ts]: Not a class, there are potential side effects of putting this in some common place within the file so just added it to the constants instead.

File: src/validation/custom.validation.ts (hunk 1 above)
Comment (mattch1) [reply src/validation/custom.validation.ts]: Added tests to focus on just this.

File: src/config/index.ts [hunk 4]
```diff
@@ -153,6 +153,7 @@ export const ENTITY_CHANGE_EMAIL = ENTITY_URL + EMAIL;
 export const ENTITY_CHANGE_LEGAL_FORM = ENTITY_URL + LEGAL_FORM;
 export const ENTITY_CHANGE_GOVERNING_LAW = ENTITY_URL + LAW_GOVERNED;
 export const ENTITY_CHANGE_PUBLIC_REGISTER = ENTITY_URL + "#is_on_register_in_country_formed_in";
+export const CONCAT_VALUE_SEPARATOR = ",";
```
Comment (mwestacott) [inline src/config/index.ts:156]: Nice. I would just suggest expanding the name a bit - CONCATENATED_VALUES_SEPARATOR.

File: src/validation/error.messages.ts [hunk 5]
```diff
@@ -97,6 +99,7 @@ export enum ErrorMessages {
...
+  MAX_ENTITY_PUBLIC_REGISTER_NAME_AND_JURISDICTION_LENGTH = "Name of register and jurisdiction must be 160 characters or less in total",
```
[the remark this answers is not in the document]
Comment (mwestacott) [reply src/validation/error.messages.ts:101]: I'd flag this up to an analyst actually @mattch1 and please record response here and/or on JIRA ticket. Thanks.

File: src/validation/error.messages.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mwestacott) [reply src/validation/error.messages.ts]: Seen that you have removed it (and associated) implementation and tests. Please just check with analyst before doing this / not testing individual field lengths. Imagine it's fine but worth checking.

File: src/config/index.ts (hunk 4 above)
Comment (mattch1) [reply src/config/index.ts]: Ok, updated this.

File: src/validation/error.messages.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (sarahocarroll) [reply src/validation/error.messages.ts]: Would this error still be needed on managing-office-corporate and beneficial-owner-other pages which have the register name field as one field, it's only the entity page that's having the fields split isn't it?

File: src/validation/error.messages.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (mattch1) [reply src/validation/error.messages.ts]: Update to 159, see comment on ticket.

File: src/validation/error.messages.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mwestacott) [reply src/validation/error.messages.ts]: Assume this is being sorted out so resolving this conversation.

**Review by mwestacott — APPROVED** [review]

**Pull request merged**